api.py

```diff
@@ -142,36 +142,113 @@ def index_lecture(self, task_id: str, source: str) -> bool:
+                chunk = block.text[min_range:max_range]
+
+                # Create a Document.
+                # TODO(ted): See if there's a way to support the LC Embedding Index abstraction that lets us use Tag here.
+                doc = Document(
+                    page_content=chunk,
+                    metadata={
+                        "source": source,
+                        "file_id": file.id,
+                        "block_id": block.id,
+                        "page": page_id
+                    },
+                )
+                documents.append(doc)
+
+        self._get_index().add_documents(documents)
+        self._update_file_status(file, "Indexed")
+        return True
+
     @post("/transcribe_lecture")
-    def transcribe_lecture(self, task_id: str, source: str):
-        file_create_task = Task.get(self.client, task_id)
-        file = File.get(self.client, json.loads(file_create_task.output)["file"]["id"])
+    def transcribe_lecture(self, file_id: str, source: str):
+        file = File.get(self.client, _id=file_id)
 
         Tag.create(self.client, file_id=file.id, kind="source", name=source)
         Tag.create(self.client, file_id=file.id, kind="title", name=YouTube(source).title)
 
         self._update_file_status(file, "Transcribing")
 
-        transcribe_lecture_task = file.blockify(self.transcriber.handle)
+        blockifier = self.client.use_plugin("s2t-blockifier-default")
+        blockify_file_task = file.blockify(blockifier.handle)
 
         return self.invoke_later(
             method="index_lecture",
-            arguments={"task_id": file_create_task.task_id, "source": source},
-            wait_on_tasks=[transcribe_lecture_task],
+            arguments={"file_id": file.id, "source": source},
+            wait_on_tasks=[blockify_file_task],
+        )
+
+    @post("/blockify_pdf")
+    def blockify_pdf(self, file_id: str, source: str):
+        file = File.get(self.client, _id=file_id)
+
+        self._update_file_status(file, "Parsing")
+
+        blockifier = self.client.use_plugin("pdf-blockifier")
+        blockify_file_task = file.blockify(blockifier.handle)
+
+        return self.invoke_later(
+            method="index_pdf",
+            arguments={"file_id": file_id, "source": source},
+            wait_on_tasks=[blockify_file_task],
         )
 
     @post("/add_lecture")
-    def add_lecture(self, youtube_url: HttpUrl) -> bool:
+    def add_lecture(self, youtube_url: HttpUrl) -> Task:
+        file_importer = self.client.use_plugin("youtube-file-importer")
         file_create_task = File.create_with_plugin(
-            self.client, plugin_instance=self.youtube_importer.handle, url=youtube_url
+            self.client, plugin_instance=file_importer.handle, url=youtube_url
         )
-        self.invoke_later(
+        return self.invoke_later(
             method="transcribe_lecture",
             arguments={"task_id": file_create_task.task_id, "source": youtube_url},
```

Comment:
I think you want to pass the file_id here since the interface of `transcribe_lecture` changed.